pkg/apis/core/validation/validation.go

```diff
@@ -9410,3 +9433,41 @@ func validateWorkloadReference(workloadRef *core.WorkloadReference, fldPath *fie
 	}
 	return allErrs
 }
+
+func validateImageVolumeStatus(imageVolumeStatus *core.ImageVolumeStatus, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	if imageVolumeStatus == nil {
+		return allErrors
+	}
+
+	if imageVolumeStatus.ImageRef == "" {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("imageRef"), imageVolumeStatus.ImageRef, "imageRef must not be empty"))
+	}
+
+	const maxImageRefLen = 256
+	if len(imageVolumeStatus.ImageRef) > maxImageRefLen {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("imageRef"), imageVolumeStatus.ImageRef, fmt.Sprintf("imageRef must not have length bigger than %d", maxImageRefLen)))
+	}
+
+	return allErrors
+}
+
+// validateVolumeStatus returns an error if the given struct has more than 1 populated field.
+func validateVolumeStatus(volumeStatus core.VolumeStatus, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	const maxStatuses = 1
+	numStatuses := 0
+
+	if volumeStatus.Image != nil {
+		numStatuses++
+		allErrors = append(allErrors, validateImageVolumeStatus(volumeStatus.Image, fldPath.Child("imageVolumeStatus"))...)
+	}
+
+	if numStatuses > maxStatuses {
+		allErrors = append(allErrors, field.Forbidden(fldPath, "may not specify more than 1 volume type"))
```

Comment:
You defined a const for max, and then embedded 1 in the error :)

I don't think you really need the const, personally.